utils.go

```diff
@@ -2,202 +2,167 @@ package loop
 
 import (
 	"context"
+	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
 	"github.com/lightninglabs/lndclient"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
 	"github.com/lightningnetwork/lnd/lnwire"
 	"github.com/lightningnetwork/lnd/routing/route"
 	"github.com/lightningnetwork/lnd/zpay32"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
 var (
-	// DefaultMaxHopHints is set to 20 as that is the default set in LND
+	// DefaultMaxHopHints is set to 20 as that is the default set in LND.
 	DefaultMaxHopHints = 20
 )
 
-// SelectHopHints is a direct port of the SelectHopHints found in lnd. It was
-// reimplemented because the current implementation in LND relies on internals
-// not externalized through the API. Hopefully in the future SelectHopHints
-// will be refactored to allow for custom data sources. It iterates through all
-// the active and public channels available and returns eligible channels.
-// Eligibility requirements are simple: does the channel have enough liquidity
-// to fulfill the request and is the node whitelisted (if specified)
-func SelectHopHints(ctx context.Context, lnd *lndclient.LndServices,
-	amtMSat btcutil.Amount, numMaxHophints int,
-	includeNodes map[route.Vertex]struct{}) ([][]zpay32.HopHint, error) {
+// isPublicNode checks if a node is public, by simply checking if there's any
+// channels reported to the node.
```

Comment:
nit: make a note here that `GetNodeInfo` will not report our private channels with a node?
If it did, this wouldn't work as a public node check.

Reply:
done